lib/streamlit/caching.py

```diff
@@ -332,10 +335,10 @@ def _read_from_mem_cache(key, allow_output_mutation):
         raise CacheKeyNotFoundError("Key not found in mem cache")
 
 
-def _write_to_mem_cache(key, value, allow_output_mutation, args_mutated):
+def _write_to_mem_cache(key, value, allow_output_mutation, args_mutated, hash_funcs):
     _mem_cache[key] = CacheEntry(
         value=value,
-        hash=None if allow_output_mutation else get_hash(value),
+        hash=None if allow_output_mutation else get_hash(value, hash_funcs=hash_funcs),
```

Comment:
Not your doing, but can you pull this `if` out of this object construction statement?

```
if allow_output_mutation:
  hash = None
else: 
  hash = get_hash(...)
```